Lib/test/test_typing.py

```diff
@@ -4228,6 +4246,46 @@ def test_get_type_hints(self):
             {'a': typing.Optional[int], 'b': int}
         )
 
+    def test_get_type_hints_generic(self):
+        self.assertEqual(
+            get_type_hints(BarGeneric[int]), 
+            {'a': typing.Optional[T], 'b': int}
+        )
+
+    def test_generic(self):
+        class A(TypedDict, Generic[T]):
```

Comment:
I would probably also add a test case for unsubscribed version as well. Something like:

```python
class WithImplicitAny(A):
   ...
```

It does not seem to be covered.

Reply:
Adding this test case caught a bug. Nice!